File: models/project.go
package models

import (
	"time"

	"gorm.io/gorm"
)

// Project 项目模型
type Project struct {
	ID          uint           `gorm:"primarykey" json:"id"`
	Name        string         `gorm:"size:255;not null" json:"name"`
	Type        string         `gorm:"size:50;not null" json:"type"` // table, show, tools, platform, origin
	LinkType    string         `gorm:"size:50" json:"link_type"`     // local, url, videoPath
	Link        string         `gorm:"size:500" json:"link"`         // 项目链接或路径
	Thumbnail   string         `gorm:"size:500" json:"thumbnail"`    // 缩略图路径
	Description string         `gorm:"type:text" json:"description"`
	SortOrder   int            `gorm:"default:0" json:"sort_order"`
	Status      int            `gorm:"default:1" json:"status"` // 1:启用 0:禁用
	CreatedAt   time.Time      `json:"created_at"`
	UpdatedAt   time.Time      `json:"updated_at"`
	DeletedAt   gorm.DeletedAt `gorm:"index" json:"-"`

	// 关联资源
	Resources []Resource `gorm:"foreignKey:ProjectID" json:"resources,omitempty"`
}

// TableName 指定表名
func (Project) TableName() string {
	return "projects"
}

File: frontend/src/api/project.ts
import request from "./request";

export interface Project {
  id?: number;
  name: string;
  type: "table" | "show" | "tools" | "platform" | "origin";
  link_type?: "local" | "url" | "videoPath";
  link?: string;
  thumbnail?: string;
  description?: string;
  sort_order?: number;
  status?: number;
  created_at?: string;
  updated_at?: string;
}

export interface ProjectListParams {
  type?: string;
  page?: number;
  page_size?: number;
}

export interface ProjectSearchParams {
  keyword?: string;
  type?: string;
  page?: number;
  page_size?: number;
}

export interface PaginationData<T> {
  list: T[];
  total: number;
  page: number;
  page_size: number;
  pages: number;
}

// 创建项目
export const createProject = (data: Project) => {
  return request.post<any, Project>("/projects", data);
};

// 获取项目列表
export const getProjects = (params: ProjectListParams) => {
  return request.get<any, PaginationData<Project>>("/projects", { params });
};

// 获取项目详情
export const getProject = (id: number) => {
  return request.get<any, Project>(`/projects/${id}`);
};

// 更新项目
export const updateProject = (id: number, data: Project) => {
  return request.put<any, Project>(`/projects/${id}`, data);
};

// 删除项目
export const deleteProject = (id: number) => {
  return request.delete(`/projects/${id}`);
};

// 更新项目排序
export const updateProjectSort = (id: number, sortOrder: number) => {
  return request.put(`/projects/${id}/sort`, { sort_order: sortOrder });
};

// 搜索项目
export const searchProjects = (params: ProjectSearchParams) => {
  return request.get<any, PaginationData<Project>>("/projects/search", {
    params,
  });
};
